content/en/docs/components/pipelines/sdk/v2/build-pipeline.md

```diff
@@ -0,0 +1,520 @@
+[input]: https://github.com/kubeflow/pipelines/blob/master/sdk/python/kfp/dsl/io_types.py
+[output]: https://github.com/kubeflow/pipelines/blob/master/sdk/python/kfp/dsl/io_types.py
+[dsl-component]: https://github.com/kubeflow/pipelines/blob/master/sdk/python/kfp/dsl/_component.py
+
+
+```python
+@component(
+    base_image='python:3.7',
+    packages_to_install=['pandas==1.1.4'],
+    output_component_file='component.yaml' # This is optional. It saves the component spec for future use.
+)
+def merge_csv(tar_data: Input[Artifact], output_csv: Output[Dataset]):
+  import glob
+  import pandas as pd
+  import tarfile
+
+  tarfile.open(name=tar_data.path, mode="r|gz").extractall('data')
+  df = pd.concat(
+      [pd.read_csv(csv_file, header=None) 
+       for csv_file in glob.glob('data/*.csv')])
+  df.to_csv(output_csv.path, index=False, header=False)
+```
+
+### Build your pipeline
+
+1.  Use [`kfp.components.load_component_from_url`][load_component_from_url]
+    to load the component specification YAML for any components that you are
+    reusing in this pipeline.
+
+[load_component_from_url]: https://kubeflow-pipelines.readthedocs.io/en/latest/source/kfp.components.html?highlight=load_component_from_url#kfp.components.load_component_from_url
+
+
+```python
+web_downloader_op = kfp.components.load_component_from_url(
+    'https://raw.githubusercontent.com/kubeflow/pipelines/master/components/web/Download/component-sdk-v2.yaml')
+```
+
+2.  Define your pipeline as a Python function. 
+
+    Your pipeline function's arguments define your pipeline's parameters. Use
+    pipeline parameters to experiment with different hyperparameters, such as
+    the learning rate used to train a model, or pass run-level inputs, such as
+    the path to an input file, into a pipeline run. The data type must be
+    specified for all pipeline parameters. 
+    
+    Use the factory functions created by
+    the `kfp.dsl.component` annotation and the
+    `kfp.components.load_component_from_url` function to create your pipeline's tasks. 
+    The inputs to the component factory functions can be pipeline parameters,
+    the outputs of other tasks, or a constant value. In this case, the
+    `web_downloader_task` task uses the `url` pipeline parameter, and the
+    `merge_csv_task` uses the `data` output of the `web_downloader_task`.
+
+    The `kfp.dsl.pipeline` annotation lets you specify the following:
+
+    *   `name`: The pipeline's name.
+    *   `description`: (Optional.) A description of the pipeline's workflow.  
+    *   `pipeline_root`: The default path where your pipeline's artifacts are
+         stored. This must be a path that your pipeline can read and write to,
+         such as a Persistent Volume Claim or a cloud service such as Google
+         Cloud Storage.
+
+
+```python
+# Define a pipeline and create a task from a component:
+@dsl.pipeline(
+    pipeline_root='gs://my-pipeline-root/example-pipeline',
```

Comment:
nit: order-wise, suggest we put `name` first, `name` is a required argument, while `pipeline_root` is optional.